### PR TITLE
Added $changeEmail() method

### PR DESCRIPTION
**Pull request by jwngr**
@katowulf - Please review but **do not merge yet!.** Wait until the JS client release goes out which adds `changeEmail()`, then merge this guy in before releasing 0.9.1. I am assuming the Firebase version will be 2.1.0, but please update the README in this PR if that version turns out to be 2.0.7.

### CODE REVIEW

**Comment by jwngr** [comment]
@katowulf - Now that Firebase 2.1.0 has been released with a `changeEmail()` method, this is ready to be merged in. So please review and merge this (and the corresponding docs changes in our website repo) as soon as possible. Once this is in, I will cut the long-awaited 0.9.1 release.

File: tests/unit/FirebaseAuth.spec.js [hunk 1]
```diff
@@ -361,42 +361,95 @@ describe('FirebaseAuth',function(){
...
+        password: '12345'
+      });
+      expect(ref.changeEmail).toHaveBeenCalledWith({
+        oldEmail: 'somebody@somewhere.com',
+        newEmail: 'otherperson@somewhere.com',
+        password: '12345'
+      }, jasmine.any(Function));
+    });
+
+    it('will reject the promise if authentication fails',function() {
```
Comment (katowulf) [inline tests/unit/FirebaseAuth.spec.js]: Is authentication the right word here? Also just below in the next unit test.

**Comment by katowulf** [comment]
This one is ready other than what looks like a copy/paste error in the spec descriptions.

**Comment by jwngr** [comment]
Back to you for merging @katowulf.

**Comment by katowulf** [comment]
yay

**Pull request merged**